### PR TITLE
`test_api_skeleton.py`: Reorg fixtures, remove pre-Python 2.7 condition, require `ruamel.yaml`

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Grouped fixtures at the top of the file. Remove pre-Python 2.7 condition. Require `ruamel.yaml` to avoid try-except.

Xref https://github.com/conda/conda-build/issues/4590

### Checklist - did you ...

- [ ] ~Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: tests/test_api_skeleton.py [hunk 1]
```diff
@@ -448,19 +453,31 @@ def test_pypi_section_order_preserved():
 
 @pytest.mark.slow
 @pytest.mark.flaky(rerun=5, reruns_delay=2)
-@pytest.mark.skipif(not external.find_executable("shellcheck"), reason="requires shellcheck >=0.7.0")
+@pytest.mark.skipif(on_win, reason="shellcheck is only available on Windows")
```
Comment (jezdez) [inline tests/test_api_skeleton.py:456]: I didn't know shellcheck is only on Windows, shouldn't that be:
```suggestion
@pytest.mark.skipif(on_win, reason="shellcheck is not available on Windows")
```

**Review by jezdez — DISMISSED** [review]
Minor nits, but nothing serious

File: tests/test_api_skeleton.py (hunk 1 above)
Comment (kenodegard) [reply tests/test_api_skeleton.py:456]: 🤦🏼‍♂️

File: tests/test_api_skeleton.py [hunk 2]
```diff
@@ -158,14 +96,111 @@ def pkginfo_pylint():
...
+
+@pytest.mark.parametrize(
+    "prefix, repo, package, version",
+    [
+        ("", "pypi", "pip", "8.1.2"),
+        ("r-", "cran", "acs", None),
+        ("r-", "cran", "https://github.com/twitter/AnomalyDetection.git", ""),
+        ("perl-", "cpan", "Moo", None),
+        ("", "rpm", "libX11-devel", None),
+        # ('lua', luarocks', 'LuaSocket', ''),
```
[the remark this answers is not in the document]
Comment (kenodegard) [reply tests/test_api_skeleton.py:150]: hm let me actually see if we can enable the test instead

[the remark this answers is not in the document]
Comment (jezdez) [reply tests/test_api_skeleton.py:150]: sure that works too

File: tests/test_api_skeleton.py (hunk 1 above)
Comment (jezdez) [reply tests/test_api_skeleton.py:456]: 🤗

File: tests/test_api_skeleton.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kenodegard) [reply tests/test_api_skeleton.py:150]: the lua skeleton feature is broken but it's a valid thing to investigate and fix or properly deprecate, I'll leave a comment with this commented-out test and open a new ticket

[the remark this answers is not in the document]
Comment (kenodegard) [reply tests/test_api_skeleton.py:150]: https://github.com/conda/conda-build/issues/4756

**Review by jezdez — APPROVED** [review]

**Pull request merged**